lib/ddtrace/profiling/preload.rb

```diff
@@ -0,0 +1,9 @@
+require 'ddtrace/profiling'
+
+if Datadog::Profiling.supported? && Datadog::Profiling.native_cpu_time_supported?
+  Datadog::Profiling::Tasks::Setup.new.execute
+# rubocop:disable Style/EmptyElse
+else
+  # TODO: Log error?
+  # puts '[DDTRACE] Profiling not supported; skipping preload.'
```

Comment:
This code runs before the entire application runs; its purpose is to patch in CPU timing before the application starts. If it fails or cannot run, it might produce an error or log message. `Datadog.configure` and the logger will not have been configured at this point.

@marcotc Should it initialize and log to the default logger or print to STDOUT? Or something else?

Reply:
I think STDOUT does the trick here, given this such an early script. I think loading all the necessary parts to make `Datadog.logger` work would be too dangerous here.

Reply:
That was my instinct too; I'll write to `STDOUT` then.